src/Microsoft.AspNetCore.Routing/Matchers/TreeMatcher.cs

```diff
@@ -137,6 +141,8 @@ internal class TreeMatcher : Matcher
 
         private Task SelectEndpointAsync(HttpContext httpContext, IEndpointFeature feature, IReadOnlyList<MatcherEndpoint> endpoints)
         {
+            var bestEndpoint = _endpointSelector.SelectBestCandidate(httpContext, endpoints);
```

Comment:
I haven't added this to any of the other matchers. Do we want to add it?

Reply:
Let's keep it here for now - I think it would create a lot of noise for the perf work that I'm doing at this stage - which is in a very early stage.